src/test/common/steps/add-children-dob-steps.js

```diff
@@ -26,22 +47,74 @@ Then(/^The back link points to the Add your children’s dates of birth page$/,
+  await assertDateOfBirthErrorPresentForChild(1)
+  await assertDateOfBirthErrorPresentForChild(2)
+})
+
+Then(/^I am informed that I need to enter a shorter name for both children$/, async function () {
+  await assertNameTooLongErrorPresentForChild(1)
+  await assertNameTooLongErrorPresentForChild(2)
+})
+
+Then(/^I am informed that I need to enter a shorter name$/, async function () {
+  await assertNameTooLongErrorPresentForChild(1)
+})
+
+async function assertNameTooLongErrorPresentForChild (index) {
+  try {
+    await assertErrorHeaderTextPresent(pages.addChildrenDOB)
+    await assertFieldErrorAndLinkTextPresentAndCorrect(
+      pages.addChildrenDOB.getChildNameFieldErrorId(index),
+      pages.addChildrenDOB.getChildNameErrorLinkCss(index),
+      'Enter a shorter name')
+  } catch (error) {
+    assert.fail(`Unexpected error caught trying to assert the too long name error present for child with index ${index} - ${error}`)
+  }
+}
+
+async function assertDateOfBirthErrorPresentForChild (index) {
+  try {
+    await assertErrorHeaderTextPresent(pages.addChildrenDOB)
+    await assertFieldErrorAndLinkTextPresentAndCorrect(
+      pages.addChildrenDOB.getChildDateOfBirthFieldErrorId(index),
+      pages.addChildrenDOB.getChildDateOfBirthErrorLinkCss(index),
+      'Enter your child’s date of birth')
+  } catch (error) {
+    assert.fail(`Unexpected error caught trying to assert the date of birth error present for child with index ${index} - ${error}`)
+  }
+}
+
+async function submitTwoSetsOfChildren3OrUnderDetails (firstChildName = 'Joe', secondChildName = 'Joanne') {
   try {
-    await pages.addChildrenDOB.enterChildUnder3Details()
+    await pages.addChildrenDOB.enterChild3OrUnderDetails(firstChildName, 0)
     await pages.addChildrenDOB.clickAddAnotherChild()
-    await pages.addChildrenDOB.enterChildUnder3Details()
+    await pages.addChildrenDOB.enterChild3OrUnderDetails(secondChildName, 0)
     await pages.addChildrenDOB.submitForm()
   } catch (error) {
     assert.fail(`Unexpected error caught trying to enter two sets of children details and submit the page - ${error}`)
   }
 }
 
-async function enterChildUnder3DetailsWithoutAName () {
+async function submitTenSetsOfChildren3OrUnderDetails () {
   try {
-    await pages.addChildrenDOB.enterChildUnder3DetailsWithoutAName()
+    var childIndex
+    for (childIndex = 1; childIndex <= 10; childIndex++) {
+      await enterChildDetailsAndClickAddAnother(childIndex)
+    }
     await pages.addChildrenDOB.submitForm()
   } catch (error) {
-    assert.fail(`Unexpected error caught trying to enter child details and submit the page - ${error}`)
+    assert.fail(`Unexpected error caught trying to enter ten sets of children details and submit the page - ${error}`)
+  }
+}
+
+async function enterChildDetailsAndClickAddAnother (childIndex) {
+  await pages.addChildrenDOB.enterChild3OrUnderDetails(`Child${childIndex}`, childIndex)
+  if (childIndex < 10) {
```

Comment:
Do we need this check here, feels like the for loop it's called from is enough to suffice there being no more than ten children.

Reply:
This is here to make sure that we don't press the Add Another Child on the final one otherwise we get a validation error on the page on submit due to the final one being empty.

Reply:
I agree with @dwybourn - this function has to have knowledge of the function that is calling it. Could this be handled with a second parameter, something like:
```
enterChildDetailsAndClickAddAnother (childIndex, shouldAdd = true) { ... }

enterChildDetailsAndClickAddAnother(childIndex, childIndex !== 10)
```

Reply:
Yes, good point @gethyn1, I've updated as you have described.